src/Client.php

```diff
@@ -40,19 +46,140 @@ public function sendRequest(RequestInterface $request)
+                }
+
+                public function __toString()
+                {
+                    try {
+                        return $this->getContents();
+                    } catch (\Throwable $e) {
+                        return '';
+                    }
+                }
+
+                public function __destruct()
+                {
+                    $this->close();
+                }
+
+                public function close()
+                {
+                    $this->cancellationTokenSource->cancel();
+                }
+
+                public function detach()
+                {
+                    // ignore
+                }
+
+                public function getSize()
+                {
+                    return null;
+                }
+
+                public function tell()
+                {
+                    return $this->position;
+                }
+
+                public function eof()
+                {
+                    return $this->eof;
+                }
+
+                public function isSeekable()
+                {
+                    return false;
+                }
+
+                public function seek($offset, $whence = SEEK_SET)
+                {
+                    throw new \RuntimeException('Stream is not seekable');
+                }
+
+                public function rewind()
+                {
+                    $this->seek(0);
+                }
+
+                public function isWritable()
+                {
+                    return false;
+                }
+
+                public function write($string)
+                {
+                    throw new \RuntimeException('Stream is not writable');
+                }
+
+                public function isReadable()
+                {
+                    return true;
+                }
+
+                public function read($length)
+                {
+                    if ($this->eof) {
+                        return '';
+                    }
+
+                    if ($this->buffer === '') {
+                        try {
+                            $this->buffer = Promise\wait($this->body->read());
+                        } catch (Artax\HttpException $e) {
+                            throw new \RuntimeException('Reading from the stream failed', 0, $e);
+                        } catch (CancelledException $e) {
+                            throw new \RuntimeException('Reading from the stream failed', 0, $e);
+                        }
+
+                        if ($this->buffer === null) {
+                            $this->eof = true;
+
+                            return '';
+                        }
+                    }
+
+                    $read = \substr($this->buffer, 0, $length);
+                    $this->buffer = (string) \substr($this->buffer, $length);
+                    $this->position += \strlen($read);
```

Comment:
not sure if we need the position as we cannot seek this stream it's also pretty useless IMO (but doesn't hurt, so don't really care)